### ISSUE TITLE
git DAG isn't being traced

### ISSUE THREAD

**Issue by esc** [AlexeySanko/pybuilder_semver_git_tag#6]
I like the project very much and I am considering including it in PyBuilder as a builtin plugin. However, I ran some tests today and unfortunately discovered, that the git DAG is being traced properly. Instead a lexicographical ordering of the tags is assumed.

Let me explain, consider the following repository structure:

![2018-03-10-181825_329x323_scrot](https://user-images.githubusercontent.com/41563/37244877-c8fec2be-248f-11e8-9d1e-ceeca4d95a7d.png)

As you can see, this is a hypothetical project that supplies two concurrent versions with updates, `1.0.0` and `2.0.0`. This is actually fairly common, especially for companies who are supplying older customers with bugfixes to older versions. 

Now, when I am on `v1` I expect the semantic version to be `1.0.1.dev`, since this is a derivative of the `1.0.0` line -- however, what I get is `2.0.1.dev`, which is clearly not what I want.

```
esc@omnitech ~/git/check-pybuilder-semver-plugin (git)-[v1] % pyb
PyBuilder version 0.11.13
Build started at 2018-03-10 18:24:33
------------------------------------------------------------
[INFO]  Found SemVer tag: 2.0.0
[INFO]  Project version was set to: 2.0.1.dev, dist_version: 2.0.1.dev20180310172434
[INFO]  Building check-pybuilder-semver-plugin version 2.0.1.dev (2.0.1.dev20180310172434)
[INFO]  Executing build in /home/esc/git/check-pybuilder-semver-plugin
[INFO]  Going to execute task publish
[INFO]  Running unit tests
[INFO]  Executing unit tests from Python modules in /home/esc/git/check-pybuilder-semver-plugin/src/unittest/python
[WARN]  No unit tests executed.
[INFO]  All unit tests passed.
[INFO]  Building distribution in /home/esc/git/check-pybuilder-semver-plugin/target/dist/check-pybuilder-semver-plugin-2.0.1.dev
[INFO]  Copying scripts to /home/esc/git/check-pybuilder-semver-plugin/target/dist/check-pybuilder-semver-plugin-2.0.1.dev/scripts
[INFO]  Writing setup.py as /home/esc/git/check-pybuilder-semver-plugin/target/dist/check-pybuilder-semver-plugin-2.0.1.dev/setup.py
[INFO]  Building binary distribution in /home/esc/git/check-pybuilder-semver-plugin/target/dist/check-pybuilder-semver-plugin-2.0.1.dev
------------------------------------------------------------
BUILD SUCCESSFUL
------------------------------------------------------------
Build Summary
             Project: check-pybuilder-semver-plugin
             Version: 2.0.1.dev (2.0.1.dev20180310172434)
      Base directory: /home/esc/git/check-pybuilder-semver-plugin
        Environments: 
               Tasks: prepare [48 ms] compile_sources [0 ms] run_unit_tests [19 ms] package [1 ms] run_integration_tests [0 ms] verify [0 ms] publish [474 ms]
Build finished at 2018-03-10 18:24:34
Build took 0 seconds (618 ms)
esc@omnitech ~/git/check-pybuilder-semver-plugin (git)-[v1] % git --no-pager log --oneline --decorate --all --graph
* 172a837c2a (HEAD -> v1) code
* 1d7d3213f9 (tag: 1.0.0) code
* 1a92803942 code
* dc5121fcf6 code
* b5bef1a6bf code
| * 9363aade62 (v2) code
| * 5ccfc2dc57 (tag: 2.0.0) code
| * bc9d7bc4f8 code
| * 7c6674d933 code
| * 8c2d78e0c3 code
| * 1f7701fc8b code
|/  
* 14d3244ee5 (tag: 0.0.1, master) zero
```
Incidentally, `git describe` does the correct thing in this case and you may want to look at that in order to fix this.

**Comment by AlexeySanko** [comment]
Hi, @esc 
I've added it to list of work.
Thanks a lot.

**Comment by AlexeySanko** [comment]
@esc , thanks a lot for contribution.
Fixed into [1.2.1](https://pypi.python.org/pypi/pybuilder-semver-git-tag/1.2.1)
